crates/engine/tree/src/tree/mod.rs

```diff
@@ -2045,6 +2099,12 @@ impl PersistenceState {
         self.rx = Some((rx, Instant::now()));
     }
 
+    /// Sets the `remove_above_state`, to the new tip number specified.
+    fn schedule_removal(&mut self, new_tip_num: u64) {
+        // TODO: what about multiple on-disk reorgs in a row?
+        self.remove_above_state = Some(new_tip_num);
```

Comment:
this could happen if there's a reorg immediately followed by another reorg while we're reorging on disk, right?

I assume what we'd need to do here is tracking the in progress `remove_above_state` and compare the `new_tip_num` against this, if the new_tip_num is lower, only then do we need a followup?

but perhaps this should instead be checked in `on_disk_reorg` function?